src/api/auks/auks_krb5_cred.c

```diff
@@ -124,6 +124,111 @@ auks_krb_is_local_tgt(krb5_principal princ, krb5_data *realm)
 		auks_krb_data_eq(princ->data[1], *realm);
 }
 
+int
+auks_krb_cc_new_unique(char ** fullname_out)
+{
+        int fstatus;
+
+        krb5_context context;
+	krb5_ccache ccache;
+
+	char *ccache_type = NULL;
+
+	/* Initialize KRB5 context */
+	fstatus = krb5_init_context(&context);
+	if (fstatus) {
+		auks_error("Error while initializing KRB5");
+		fstatus = AUKS_ERROR_KRB5_CRED_INIT_CTX;
+		return fstatus;
+	}
+
+	/* Get the default ccache */
+	fstatus = krb5_cc_default(context, &ccache);
+	if (fstatus) {
+	        auks_error("Error while getting default ccache");
+		fstatus = AUKS_ERROR_KRB5_CRED_OPEN_CC ;
+		goto out;
+	}
+
+	/* Get the default ccache type */
+	ccache_type = krb5_cc_get_type(context, ccache);
+
+	/* Generate a new unique ccache */
+	fstatus = krb5_cc_new_unique(context, ccache_type, NULL, &ccache);
+	if (fstatus) {
+		auks_error("Error while generating new unique ccache of type %s", ccache_type);
+		fstatus = AUKS_ERROR_KRB5_CRED_OPEN_CC ;
+		goto out;
+	}
+
+
+	fstatus = krb5_cc_get_full_name(context, ccache, fullname_out);
+	if (fstatus) {
+		auks_error("Error while geting ccache full name");
+		fstatus = AUKS_ERROR_KRB5_CRED_OPEN_CC ;
+		goto out;
+	}
+
+
+	/* Call krb5_cc_switch for ccache that supports if */
+	if (krb5_cc_support_switch(context, ccache_type)) {
+		fstatus = krb5_cc_switch(context, ccache);
+		if (fstatus) {
+			auks_error("Error while calling krb5_cc_switch");
+			fstatus = AUKS_ERROR_KRB5_CRED_OPEN_CC ;
+			goto out;
+		}
+	}
+
+ out:
+	/* Close ccache handles */
+	if (ccache != NULL)
```

Comment:
I am not sure that ccache is set to NULL in the libkrb5 in case of error during resolve op, I am not sure this is the right way to handle conditional cleaning. It might work, if krb5_ccache is a pointer and if the compiler explicitely set to 0 any undefined variables. I would not bet on the second to stay true everytime :)
You can have a look at the auks_krb5_cred_store() to see the closing logic to properly handle krb5_ccache.